lib/manager/components/NoteForm.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+
+import Icon from '@conveyal/woonerf/components/icon'
+import React, {Component} from 'react'
+import { Button, Checkbox, Col, FormControl, Media, Panel } from 'react-bootstrap'
+
+import {getComponentMessages} from '../../common/util/config'
+import { getProfileLink } from '../../common/util/util'
+import type {Feed} from '../../types'
+import type {ManagerUserState} from '../../types/reducers'
+
+type Props = {
+  feedSource: Feed,
+  newNotePosted: ({body: string}) => void,
+  stacked?: boolean,
+  user: ManagerUserState
+}
+
+type State = {
+  adminOnly: boolean,
+  body: string
+}
+
+const DEFAULT_STATE = {
+  adminOnly: false,
+  body: ''
+}
+
+export default class NoteForm extends Component<Props, State> {
+  messages = getComponentMessages('NoteForm')
+  state = DEFAULT_STATE
+
+  _onChange = ({target}: {target: HTMLInputElement}) =>
+    this.setState({[target.name]: target.type === 'checkbox' ? target.checked : target.value})
+
+  _onClickPublish = () => {
+    this.props.newNotePosted(this.state)
+    this.setState(DEFAULT_STATE)
+  }
+
+  render () {
+    const {
+      feedSource,
+      stacked,
+      user
+    } = this.props
+    const {Body, Heading, Left} = Media
+    const {adminOnly, body} = this.state
+    const {profile} = user
+    if (!profile) {
+      console.warn('Could not locate user profile', user)
+      return null
+    }
+    const isProjectAdmin = user.permissions && user.permissions.isProjectAdmin(
+      feedSource.projectId, feedSource.organizationId
+    )
+    const userLink = user
+      ? <a href={getProfileLink(profile.email)}>{profile.email}</a>
```

Comment:
This generated a link that sent me to a 404 page. Might be worth refactoring `getProfileLink` to return null if the link won't be valid and then render a span instead?

Reply:
**Good catch, but tldr I think this is out of scope.**

This could get complicated, so I think's it's probably out of scope (this isn't really new code, just pulled out of NotesViewer). Also, rendering just a span will not allow the user to click through and potentially create a profile image if they're so inclined (though I doubt they will care).

In any case, the profile link, whether valid or invalid, triggers a redirect to either the valid gravatar URL (200) or the 404 page.

Here's my profile link: https://en.gravatar.com/466d1e90516eab20031b225df44decdd

Inspect the requests to see it in action. Handling all of this in our codebase doesn't sound like a great idea haha. There might be a more elegant way to go about it, but I'm not sure it's worth the time.